### PR TITLE
enable eloquent rcpputils pointer traits

### LINKED ISSUE #49: enable rcpputils pointer traits
top level issue for leveraging `rcpputils::is_pointer` which in addition to `std::is_pointer` checks for smart pointers as well.

### PR DESCRIPTION
**Pull request by Karsten1987**
fixes #49 

Signed-off-by: Karsten Knese <karsten@openrobotics.org>

### CODE REVIEW

File: hardware_interface/CMakeLists.txt [hunk 1]
```diff
@@ -27,6 +28,7 @@ target_include_directories(
   include)
 ament_target_dependencies(
   hardware_interface
+  "rcpputils"
```
Comment (bmagyar) [inline hardware_interface/CMakeLists.txt]: why the quotation marks around these ones?

File: hardware_interface/CMakeLists.txt [hunk 2]
```diff
@@ -52,6 +54,7 @@ if(BUILD_TESTING)
 
   ament_add_gtest(test_macros test/test_macros.cpp)
   target_include_directories(test_macros PRIVATE include)
+  ament_target_dependencies(test_macros "rcpputils")
```
Comment (bmagyar) [inline hardware_interface/CMakeLists.txt]: this is only relevant when building rcpputils from source in the same workspace. Unless there's any other reason I'd opt for not cluttering the cmake more

File: hardware_interface/CMakeLists.txt (hunk 1 above)
Comment (Karsten1987) [reply hardware_interface/CMakeLists.txt]: That’s basically ROS2 convention and is used throughout that repo as such.

File: hardware_interface/CMakeLists.txt (hunk 2 above)
Comment (Karsten1987) [reply hardware_interface/CMakeLists.txt]: What? I don’t understand this.
The test is relying on the ‘macros.hpp’ which itself depends on ‘rcpputils’. There is no target interface for this, therefore the dependency has to be attached to the test as well as exported for other packages to be usable.

Comment (bmagyar) [reply hardware_interface/CMakeLists.txt]: ah sorry, I misread it as `add_dependencies`, forget it

**Pull request merged**